source/config/profileUpgradeSteps.py

```diff
@@ -134,3 +139,160 @@ def upgradeConfigFrom_7_to_8(profile: Dict[str, str]) -> None:
 			profile['documentFormatting']['reportTableHeaders'] = ReportTableHeaders.ROWS_AND_COLUMNS.value
 		else:
 			profile['documentFormatting']['reportTableHeaders'] = ReportTableHeaders.OFF.value
+
+
+def upgradeConfigFrom_8_to_9(profile: Dict[str, str]) -> None:

```

Comment:
It appears to be relatively easy to do this
https://stackoverflow.com/a/53845083/8030743

Note, I haven't tested this and it would be good to place this type in a more general location
```py
from __future__ import annotations 
configProfileT = Dict[str, Union[str, "configProfileT"]]
```

Reply:
@CyrilleB79 - can you make this change?

Reply:
profile is of type `configobj.ConfigObj`, which is a child class of `dict`.
I do not understand why we used `Dict[str, str])` for type hint, and why we need to specify a dedicated recursive type for type hints instead of using `configobj.ConfigObj` instead.

Maybe I missed something about type hints?

Reply:
Oh, I'm not sure why it was introduced as that to begin with then. Could you use the real type `configobj.ConfigObj`? Thanks for picking this up